docs/puzzles/pooling.md

```diff
@@ -0,0 +1,320 @@
+
+We are also calculating one additional piece of information.  Let's look at it now:
+
+```chialisp
+(defun-inline calculate_pool_reward (pool_reward_height P2_SINGLETON_PUZZLE_HASH POOL_REWARD_PREFIX pool_reward_amount)
+  (sha256 (logior POOL_REWARD_PREFIX (logand (- (lsh (q . 1) (q . 128)) (q . 1)) pool_reward_height)) P2_SINGLETON_PUZZLE_HASH pool_reward_amount)
+)
+```
+
+This line may look complex, but it's doing something fairly simplistic.  It is calculating the coin ID of the reward coin it is claiming by manually calculating the parent info from the `POOL_REWARD_PREFIX` and `pool_reward_height`.
+
+Why do we have to do this? The manual calculation is due to the fact that this singleton may have other payments made to it.  Right now, since we are lending our singleton to the pool, we don't want the singleton to be able to claim those rewards. We specifically only want the pool to be able to claim pool rewards that are generated from farming.  Since we know these rewards have a `parent_coin_info` that is a [special format](https://chialisp.com/docs/coin_lifecycle#farming-rewards), we can manually calculate it to ensure that the pool can't lie to us and pass in the ID of a non-reward coin.
+
+Let's take a look at this section here:
+
+```chialisp
+(logior POOL_REWARD_PREFIX (logand (- (lsh (q . 1) (q . 128)) (q . 1)) pool_reward_height))
+```
+
+`(- (lsh (q . 1) (q . 128)) (q . 1))` is simply a way of generating a sequence of 32 `f`s. We then use `logand` on that with the pool reward height, which in an honest scenario should do absolutely nothing to it (more on this in a minute).  Finally, we use `logior` to combine the two values into one string.  Let's say we have a height of `abcdef`.  Our final product will be `ccd5bb71183532bff220ba46c268991a00000000000000000000000000abcdef`.
+
+Why do we need this extra `logand` with the string of `f`s?  It's to prevent a relatively obscure, but possible attack.  Remember that an attacker can pass whatever they want in through the solution.  For example, a 32 byte hex string.  If they passed through the right hex string, they could completely control the output of our calculation *except* for the bits that happen to be set to 1 in the genesis challenge half of the `POOL_REWARD_PREFIX` (62/128 of them).  The idea is that a pool could grind out a coin whose parent ID has all of those bits set, create the coin, and then use the singleton to claim it.  Why claim a coin that you already had control of? This will become more apparent in the waiting room puzzle, but they could hypothetically "freeze" the singleton in their pool if they were able to reset the puzzle.  It's not as important in the pool member puzzle, but it is still probably best to prevent the pool from spending the singleton in an unwanted way.  The `logand` ensures that only bits on the right can change, every bit on left gets set to 0 before it is evaluated with the `logior`.
```

Comment:
```suggestion
Why do we need this extra `logand` with the string of `f`s?  It's to prevent a relatively obscure, but possible attack.  Remember that an attacker can pass whatever they want in through the solution.  For example, a 32 byte hex string.  If they passed through the right hex string, they could completely control the output of our calculation *except* for the bits that happen to be set to 1 in the genesis challenge half of the `POOL_REWARD_PREFIX` (62/128 of them).  The idea is that a pool could grind out a coin whose parent ID has all of those bits set, create the coin, and then use the singleton to claim it.  Why claim a coin that you already had control of? This will become more apparent in the waiting room puzzle, but they could hypothetically "freeze" the singleton in their pool if they were able to reset the puzzle.  It's not as important in the pool member puzzle, but it is still probably best to prevent the pool from spending the singleton in an unintended way.  The `logand` ensures that only bits on the right can change, every bit on left gets set to 0 before it is evaluated with the `logior`.
```